src/audio_sdl.h

```diff
@@ -27,6 +28,13 @@ class SdlAudio : public GenericAudio {
 
 	void LockMutex() const override;
 	void UnlockMutex() const override;
+	void LockMidiOutMutex() const override;
+	void UnlockMidiOutMutex() const override;
+
+	bool midiout_thread_exit = false;
```

Comment:
`std::atomic<bool>`? Also, replace assignments with explicit calls to `load` and `store` to make it obvious that multi-threaded synchronization is happening.